types/mkdirp/index.d.ts

```diff
@@ -35,6 +53,46 @@ declare namespace mkdirp {
         fs?: FsImplementationSync;
     }
 
-    function sync(dir: string, opts?: Mode | OptionsSync): Made;
+    /**
+     * Synchronously create a new directory and any necessary subdirectories at
+     * dir with octal permission string opts.mode. If opts is a string or number,
+     * it will be treated as the opts.mode. If opts.mode isn't specified, it
+     * defaults to 0o777 & (~process.umask()).
+     * You can optionally pass in an alternate fs implementation by passing in
+     * opts.fs. Your implementation should have opts.fs.mkdirSync(path, mode)
+     * and opts.fs.statSync(path).  You can also override just one or the other
+     * of mkdirSync and statSync by passing in opts.statSync or opts.mkdirSync,
+     * or providing an fs option that only overrides one of these.
+     * @returns Returns the first directory that had to be created, or undefined if everything already exists.
+     */
+    function sync(dir: string, opts?: Mode | Options): string|null;
+
+    /**
+     * Use the manual implementation (not the native one). This is the default
+     * when the native implementation is not available or the stat/mkdir
+     * implementation is overridden.
+     */
+    function manual(dir: string, opts?: Mode | Options): Promise<string|undefined>;
+
+    /**
+     * Use the manual implementation (not the native one). This is the default
+     * when the native implementation is not available or the stat/mkdir
+     * implementation is overridden.
+     */
+    function manualSync(dir: string, opts?: Mode | Options): string|null;
+
+    /**
+     * Use the native implementation (not the manual one). This is the default
+     * when the native implementation is available and stat/mkdir are not
+     * overridden.
+     */
+    function native(dir: string, opts?: Mode | Options): Promise<string|undefined>;
+
+    /**
+     * Use the native implementation (not the manual one). This is the default
+     * when the native implementation is available and stat/mkdir are not
+     * overridden.
+     */
+    function nativeSync(dir: string, opts?: Mode | Options): string|null;
```

Comment:
https://github.com/isaacs/node-mkdirp/blob/7cd4e9eb9a4376b551414a154b26697cf21d9b93/lib/mkdirp-native.js#L33

The `mkdirpManualSync` method returns `string` or `undefined`.

The `findMade` too:

* https://github.com/isaacs/node-mkdirp/blob/7cd4e9eb9a4376b551414a154b26697cf21d9b93/lib/mkdirp-native.js#L27
* https://github.com/isaacs/node-mkdirp/blob/7cd4e9eb9a4376b551414a154b26697cf21d9b93/lib/find-made.js#L16